### ISSUE TITLE
Issue in hasFormat method

### ISSUE THREAD

**Issue by roberto910907** [briannesbitt/Carbon#1497]
Hello,

I encountered an issue with the following code:
```php
Carbon::hasFormat('10-02-31', 'Y-m-d');
Carbon::hasFormat('19-08-21', 'Y-m-d');
```
Carbon version: **1.32.0**
PHP version: **PHP 7.2.11-3+ubuntu18.04.1+deb.sury.org+1**

I expected to get:
```
false
```

But I actually get:
```
true
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

I confirm this as an unexpected result, I will try to plan a fix.

**Comment by kylekatarnls** [comment]
Hi again, we will fix it as a patch for our latest major version. It means it will be fixed in **2.5.1** and **1.34.1**. We avoid to release too many patches on previous minor releases except for major bugs or security issues.

I hope it would be fine for you to update to **1.34.1** when it's released.

Else, you can still propose a pull-request on version-1.32 branch to apply the fix in 1.32.1 and in this case explaining shortly why you cannot move to 1.34.x would help us too.
